src/main/java/org/apache/maven/plugins/enforcer/MaxDependencyUpdates.java

```diff
@@ -0,0 +1,366 @@
+     *
+     * @since 2.14.0
+     */
+    protected List<String> pluginDependencyExcludes = emptyList();
+
+    /**
+     * List of plugin management dependency inclusion patterns.
+     * Only dependencies matching all the patterns will be considered.
+     * The wildcard "*" can be used as the only, first, last or both characters in each token.
+     * The version token does support version ranges.
+     *
+     * @since 2.14.0
+     */
+    protected List<String> pluginManagementDependencyIncludes = singletonList( WILDCARD );
+
+    /**
+     * List of plugin dependency management exclusion patterns.
+     * Only dependencies matching none of the patterns will be considered.
+     * The wildcard "*" can be used as the only, first, last or both characters in each token.
+     * The version token does support version ranges.
+     *
+     * @since 2.12.0
+     */
+    protected List<String> pluginManagementDependencyExcludes = emptyList();
+
+    /**
+     * Retrieves the maven project from metadata
+     * @param ruleHelper EnforcerRuleHelper object
+     * @return maven project
+     */
+    private static MavenProject getMavenProject( EnforcerRuleHelper ruleHelper )
+    {
+        try
+        {
+            return (MavenProject) ruleHelper.evaluate( "${project}" );
+        }
+        catch ( ExpressionEvaluationException e )
+        {
+            throw new RuntimeException( "Cannot evaluate project metadata", e );
+        }
+    }
+
+    /**
+     * Creates the VersionsHelper object
+     * @param ruleHelper EnforcerRuleHelper object
+     * @return VersionsHelper object
+     */
+    @SuppressWarnings( "unchecked" )
+    private static VersionsHelper createVersionsHelper( EnforcerRuleHelper ruleHelper )
+    {
+        try
+        {
+            return new DefaultVersionsHelper.Builder()
+                    .withRepositorySystem( ruleHelper.getComponent( RepositorySystem.class ) )
+                    .withArtifactResolver( ruleHelper.getComponent( ArtifactResolver.class ) )
+                    .withArtifactMetadataSource( ruleHelper.getComponent( ArtifactMetadataSource.class ) )
+                    .withRemoteArtifactRepositories( (List<ArtifactRepository>)
+                            ruleHelper.evaluate( "${project.remoteArtifactRepositories}" ) )
+                    .withRemotePluginRepositories( (List<ArtifactRepository>)
+                            ruleHelper.evaluate( "${project.remotePluginRepositories}" ) )
+                    .withLocalRepository( (ArtifactRepository) ruleHelper.evaluate( "${localRepository}" ) )
+                    .withWagonManager( ruleHelper.getComponent( WagonManager.class ) )
+                    .withSettings( (Settings) ruleHelper.evaluate( "${settings}" ) )
+                    .withServerId( (String) ruleHelper.evaluate( "${maven.version.rules.serverId}" ) )
+                    .withRulesUri( null )
+                    .withRuleSet( null )
```

Comment:
RuleSet can be useful ...

Reply:
I'll add it. 

It's actually extremely easy to add all the needed options, we can really go with the enforcer plugin.